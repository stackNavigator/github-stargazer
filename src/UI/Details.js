import React from 'react'

import { DetailsView, HeaderText, DetailsText } from '../Styled'

const Details = ({ headerCaption, mainInfo }) => (
  <DetailsView>
    <HeaderText>{headerCaption}</HeaderText>
    {mainInfo && Object.entries(mainInfo).map(([key, value]) => (
      <DetailsText key={key}>{`${key}: ${value}`}</DetailsText>
    )
    )}
  </DetailsView>
)

export default Details