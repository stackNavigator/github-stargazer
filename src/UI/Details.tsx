import React from 'react';

import {DetailsView, HeaderText, DetailsText} from '../Styled';

interface Props {
  headerCaption: string;
  mainInfo?: {};
}

const Details = ({headerCaption, mainInfo}: Props) => (
  <DetailsView>
    <HeaderText>{headerCaption}</HeaderText>
    {mainInfo &&
      Object.entries(mainInfo).map(([key, value]) => (
        <DetailsText key={key}>{`${key}: ${value}`}</DetailsText>
      ))}
  </DetailsView>
);

export default Details;
