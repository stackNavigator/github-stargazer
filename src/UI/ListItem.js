import React from 'react'
import { Text } from 'react-native'

import { ListItemView, ListItemText } from '../Styled'

const ListItem = ({ leftCaption, rightCaption }) => (
  <ListItemView>
    <ListItemText>{leftCaption}</ListItemText>
    <ListItemText>{rightCaption}</ListItemText>
  </ListItemView>
)

export default ListItem