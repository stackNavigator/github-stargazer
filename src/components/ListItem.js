import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { } from 'react-redux'
import Swipeout from 'react-native-swipeout'

import { ListItemView, InfoText } from '../Styled'

const ListItem = ({ leftCaption, rightCaption }) => {
  const { navigate } = useNavigation()
  return (
    <Swipeout right={[{
      text: 'Delete',
      backgroundColor: 'red',
      underlayColor: 'rgba(0, 0, 0, 1, 0.6)',
      onPress: () => alert('Works!')
    }]}
    autoClose={true}
    backgroundColor="transparent" >
      <ListItemView onPress={() => navigate('RepoDetails', { name: leftCaption })}>
        <InfoText>{leftCaption}</InfoText>
        <InfoText>{rightCaption}</InfoText>
      </ListItemView>
    </Swipeout>
  )
}

export default ListItem