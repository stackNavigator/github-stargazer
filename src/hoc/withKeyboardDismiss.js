import React from 'react'
import { TouchableWithoutFeedback, Keyboard, KeyboardAvoidingView } from 'react-native'

const withKeyboardDismiss = WrappedComponent => props => (
  <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
    <KeyboardAvoidingView behavior="padding">
      <WrappedComponent {...props} />
    </KeyboardAvoidingView>
  </TouchableWithoutFeedback>
)

export default withKeyboardDismiss