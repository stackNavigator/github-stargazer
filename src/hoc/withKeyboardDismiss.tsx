import React from 'react';
import {
  TouchableWithoutFeedback,
  Keyboard,
  KeyboardAvoidingView,
} from 'react-native';

const withKeyboardDismiss = (WrappedComponent: any) => (props: any) => (
  <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
    <KeyboardAvoidingView behavior="padding">
      <WrappedComponent {...props} />
    </KeyboardAvoidingView>
  </TouchableWithoutFeedback>
);

export default withKeyboardDismiss;
