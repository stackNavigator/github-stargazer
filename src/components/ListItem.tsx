import React, {useContext} from 'react';
import {useNavigation} from '@react-navigation/native';
import Swipeout from 'react-native-swipeout';

import {StoreContext} from '../hoc/withStore';
import {ListItemView, InfoText} from '../Styled';
import {IStore} from '../store';

const ListItem = ({leftCaption, rightCaption}) => {
  const {navigate} = useNavigation();
  const store: IStore = useContext(StoreContext);
  return (
    <Swipeout
      right={[
        {
          text: 'Delete',
          backgroundColor: 'red',
          underlayColor: 'rgba(0, 0, 0, 1, 0.6)',
          onPress: () => store.removeRepo(leftCaption),
        },
      ]}
      autoClose={true}
      backgroundColor="transparent">
      <ListItemView
        onPress={() => navigate('RepoDetails', {name: leftCaption})}>
        <InfoText>{leftCaption}</InfoText>
        <InfoText>{rightCaption}</InfoText>
      </ListItemView>
    </Swipeout>
  );
};

export default ListItem;
