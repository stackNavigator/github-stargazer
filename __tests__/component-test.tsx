import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import Details from '../src/UI/Details';
import ListItem from '../src/components/ListItem';
import withKeyboardDismiss from '../src/hoc/withKeyboardDismiss';

jest.mock('@react-navigation/native', () => ({
  useRoute: jest.fn(() => ({params: {name: 'default/name'}})),
  useNavigation: () => ({navigate: jest.fn()}),
}));
React.useContext = jest.fn();
React.createContext = jest.fn();

describe('Details component', () => {
  it('renders correctly without info prop', () => {
    const component = renderer
      .create(<Details headerCaption="Test Header" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('ListItem component', () => {
  it('renders full_name and stars count correctly', () => {
    const component = renderer
      .create(<ListItem leftCaption="default/repo" rightCaption="15001" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('withKeyboardDismiss HOC', () => {
  it('component is wrapped correctly', () => {
    const HOC = withKeyboardDismiss(Details);
    const component = renderer.create(<HOC />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
