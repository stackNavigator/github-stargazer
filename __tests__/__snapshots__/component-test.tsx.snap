// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders correctly without info prop 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 25,
          "fontWeight": "700",
          "marginBottom": 20,
        },
      ]
    }
  >
    Test Header
  </Text>
</View>
`;

exports[`ListItem component renders full_name and stars count correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dbddde",
        "overflow": "hidden",
      },
      undefined,
      Array [
        Object {
          "backgroundColor": "transparent",
        },
      ],
    ]
  }
>
  <View
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {},
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 20,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
            },
          ]
        }
      >
        default/repo
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 20,
            },
          ]
        }
      >
        15001
      </Text>
    </View>
  </View>
  <View />
  <View />
</View>
`;

exports[`withKeyboardDismiss HOC component is wrapped correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 25,
            "fontWeight": "700",
            "marginBottom": 20,
          },
        ]
      }
    />
  </View>
</View>
`;
